### ISSUE TITLE
Using definitions with binding together does not seem to work.

### ISSUE THREAD

**Issue by FrankCoutinho** [jdbi/jdbi#1547]
Even though I pass ASC or DESC in the request.getOrder() it does not work. Am I doing something wrong?

`
MAIN_QUERY_TEMPLATE = "SELECT id, name, locale FROM field ORDER BY :column <ordering> LIMIT :limit OFFSET :offset ";
`

public List<Field> execute(PaginatedRequest request) {

        return jdbi.withHandle(
            handle -> handle
                .createQuery(MAIN_QUERY_TEMPLATE)
                .define("ordering", request.getOrder())
                .bind("limit", request.getPageSize())
                .bind("column", request.getSortBy())
                .bind("offset", request.getPageSize() * (request.getPageNumber() - 1))
                .map(
                    (rs, ctx) -> new Field(
                        rs.getString("id"),
                        rs.getString("name"),
                        rs.getString("locale")))
                .list()
        );
}`

**Comment by TheRealMarnes** [comment]
Three things come to mind:
- which TemplateEngine is active on the handle?
- are you sure `request.getOrder` returns a value that can be rendered correctly? IIRC the passed object is simply `toString`ed by the default TemplateEngine
- define "not working", can you capture the final rendered query that's sent to the db?

**Comment by FrankCoutinho** [comment]
I was using the default template engine. I found out the problem:

-  I believe when you bind a string parameter (:parameter) it gets transforms to 'parameterValue' and when you use it in a ORDER BY clause in Postgres doest not work as intended.

TL;DR I had to change from `:column` to `<column>`

**Comment by TheRealMarnes** [comment]
Oh yes, indeed, JDBC does not allow specifying the COL in `order by COL ORDER` as a query parameter. You have to use templating for both COL and ORDER.

**Comment by qualidafial** [comment]
Please be careful to sanitize values you define for the template engine. This is a common vector for SQL injection attacks.

**Comment by qualidafial** [comment]
I suggest using an enum for your column name attribute to protect against injection.
